design_pim/overview.rst

```diff
@@ -0,0 +1,354 @@
+
+As we are an open source company, our product can used and customized for a lot of different reasons and by a lot of different people (integrators, clients, contributors, technological partners). With this in mind, it was impossible to base our architecture on overrides to customize it.
+
+We decided to create a tree based architecture where each form of the application would be a tree of extensions. If you want to change a part of a page or add something to it, you need to create a RequireJS module and register it in the tree of form extensions.
+
+The form extension
+++++++++++++++++++
+
+To continue on our example we can use our previously created Backbone view to transform it in a form extension.
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/new_view.js
+    define(
+        ['pim/form'],
+        function (BaseForm) {
+            return BaseForm.extend({
+                events: {
+                    'click': 'clicked'
+                },
+
+                render: function () {
+                    this.$el.html('<div>Hello world</div>')
+                },
+
+                clicked: function (event) {
+                    console.log(event);
+                }
+            });
+        }
+    );
+
+
+As you can see, we didn't changed much: we now extend the BaseForm instead of the Backbone.View. As BaseForm extends itself the ``Backbone.View``, everything works as before and you can use all Backbone features.
+
+Register it
++++++++++++
+
+Now that we created our form extension we need to register it:
+
+.. code-block:: yaml
+
+    # src/Acme/Bundle/CustomBundle/Resources/config/form_extensions.yml
+    extensions:
+        my_form_extension:
+            module: my_custom_view             # Your RequireJS module name
+            parent: pim-product-edit-form-meta # The parent of your extension (the meta section of the product edit form in this case)
+
+
+A fiew words about this small configuration:
+
+- the key ``my_form_extension`` should be a unique key to represent your form extension.
+- you can declare multiple form extensions with the same RequireJS module.
+- to be registered your configuration file should be named ``form_extensions.yml`` or putted in a ``form_extensions`` folder in your bundle ``Resources/config`` folder.
+- You can override any form extension by using the same unique key (the order of the override is defined by your ``AppKernel.php`` registration order).
+
+This configuration is the minimal example. Here are the other parameters that you can use:
+
+.. code-block:: yaml
+
+    # src/Acme/Bundle/CustomBundle/Resources/config/form_extensions.yml
+    extensions:
+        my_form_extension:
+            module: my_custom_view
+            parent: pim-product-edit-form-meta
+            targetZone: header       # Where to put the extension in the parent DOM
+            position: 90             # Its position among the other extensions
```

Comment:
[optional] .. etc

Reply:
In fact they are all optional and I explain just above that this is a minimal example. I don't know if I need to add it here again.